### PR TITLE
modulartokenizer added special tokens + minor changes

### PR DESCRIPTION
**Pull request by floccinauc**

### CODE REVIEW

File: fusedrug/data/tokenizer/modulartokenizer/README.md [hunk 1]
```diff
@@ -83,6 +83,10 @@ Additional tokens that are added to the newly created tokenizer are defined in [
 An example of usage of the modular tokenizer is found here: [ModularTokenizer use](https://github.com/BiomedSciAI/fuse-drug/blob/a1b9564eb54b9fe39890645fb5378c13aedde6fb/fusedrug/data/tokenizer/modulartokenizer/test_multi_tokenizer_use.py#L16). It uses the same config as the creation example, and loads the jsons that were saved by the creation code.
 
 The example loads a ready ModularTokenizer stored here: [Word AA + BPE SMILES, path load](https://github.com/BiomedSciAI/fuse-drug/tree/main/fusedrug/data/tokenizer/modulartokenizer/pretrained_tokenizers/modular_AA_SMILES_single_path) using load(), i.e. no specific json paths need to be specified, and the load config is automatically generated by save().
+
+### Adding special tokens:
+An example of adding special tokens to an existing tokenizer is found here: [ModularTokenizer update]()
```
Comment (mosheraboh) [inline fusedrug/data/tokenizer/modulartokenizer/README.md:88]: Where 🤔 ?

File: fusedrug/data/tokenizer/modulartokenizer/modular_tokenizer.py [hunk 2]
```diff
@@ -1023,13 +1033,16 @@ def update_vocab(
                 f"Trying to add to the tokenizer tokens that are currently regular tokens in the tokenizer. Choose other token names. Conflicting tokens are {tokens_regular}"
             )
 
+        if len(tokens) == 0:
+            return
```
Comment (mosheraboh) [inline fusedrug/data/tokenizer/modulartokenizer/modular_tokenizer.py:1037]: Should you return 0 in that case?

File: fusedrug/data/tokenizer/ops/test_modular_tokenizer_ops.py [hunk 3]
```diff
@@ -16,25 +14,25 @@ def seed(seed_value: int) -> int:
...
-        tokenizer_path=tokenizer_path,
-        max_size=encoder_inputs_max_seq_len,
-        pad_token="<PAD>",
-    )
-    labels_tokenizer_op = FastTokenizer(
-        tokenizer_path=tokenizer_path,
-        max_size=labels_inputs_max_seq_len,
-        pad_token="<PAD>",
-    )
+# def tokenizer(
```
Comment (mosheraboh) [inline fusedrug/data/tokenizer/ops/test_modular_tokenizer_ops.py:17]: Delete if it's not required anymore

**Review by mosheraboh — APPROVED** [review]
Looks good.
See minor comments inline

**Review by mosheraboh — APPROVED** [review]
Looks good.
See minor comments inline

File: fusedrug/data/tokenizer/modulartokenizer/README.md (hunk 1 above)
Comment (floccinauc) [reply fusedrug/data/tokenizer/modulartokenizer/README.md:88]: Will update and add links to main once merged

File: fusedrug/data/tokenizer/ops/test_modular_tokenizer_ops.py (hunk 3 above)
Comment (floccinauc) [reply fusedrug/data/tokenizer/ops/test_modular_tokenizer_ops.py:17]: Done

File: fusedrug/data/tokenizer/modulartokenizer/modular_tokenizer.py (hunk 2 above)
Comment (floccinauc) [reply fusedrug/data/tokenizer/modulartokenizer/modular_tokenizer.py:1037]: Yes. good catch. Fixed.

**Pull request merged**